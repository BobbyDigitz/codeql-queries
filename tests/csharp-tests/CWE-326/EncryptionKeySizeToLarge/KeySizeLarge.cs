using System;
using System.IO;
using System.Text;
using System.Runtime.Serialization;
using System.Security.Cryptography;

public class StaticSalt
{
    public void Test1()
    {
        // BAD
        var dsa1 = new DSACryptoServiceProvider(2042);

        // GOOD
        var dsa2 = new DSACryptoServiceProvider(1024);

    }
}