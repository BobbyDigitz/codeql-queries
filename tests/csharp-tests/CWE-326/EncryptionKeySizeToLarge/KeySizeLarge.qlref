CWE-326/EncryptionKeySizeToLarge.ql