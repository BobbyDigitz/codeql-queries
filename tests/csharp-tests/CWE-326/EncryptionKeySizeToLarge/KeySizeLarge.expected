| KeySizeLarge.cs:12:20:12:53 | object creation of type DSACryptoServiceProvider | Key size 2042 is to large for algorithm (max: 1024) |
